src/test/lib/statistics/cardinality_estimator_test.cpp

```diff
@@ -913,4 +914,150 @@ TEST_F(CardinalityEstimatorTest, NonQueryNodes) {
   EXPECT_EQ(estimator.estimate_cardinality(DummyTableNode::make()), 0.0f);
 }
 
+// Usually, we assume that we know nothing about predicates getting their values from uncorrelated subqueries and we
+// simply forward the input estimations. However, special predicates with subqueries can stem from join rewrites (e.g.,
+// JoinToPredicateRewriteRule). We estimate the resulting plans just like we would have done it for a semi-join. The
+// following two test cases ensure the correct behavior for the possible rewrites.
+TEST_F(CardinalityEstimatorTest, ValueScanWithUncorrelatedSubquery) {
+  // Case (i): Predicate column = <subquery>
+  // Example query:
+  //     SELECT n_name FROM nation WHERE n_regionkey = (SELECT r_regionkey FROM region WHERE r_name = 'ASIA');
```

Comment:
Add `Note: yadda yadda is a selection on the primary key. We know ... single line` ?